vms/proposervm/vm.go

```diff
@@ -414,7 +405,7 @@ func (vm *VM) repair(ctx context.Context) error {
 		}
 	case block.ErrIndexIncomplete:
 	default:
-		return nil
+		return err
```

Comment:
@abi87 can you look into this? Seems like this should have been returning an error before

Reply:
May also want to add a regression test.

Reply:
@StephenButtolph  see [here](https://github.com/ava-labs/avalanchego/pull/1915)